### DISCUSSION TITLE
Customize search

### DISCUSSION THREAD

**Discussion by ghost** [sa-si-dev/virtual-select#145] [Q&A] [answered]
Hii,
I need little help in customizing the search feature in virtual-select.
The requirement is to match the first 3 characters of the options. 
Can anyone please help me with this ?? 🙏🙏🙏

**Comment by sa-si-dev** [comment]
Could you please share an example or scenario to understand your requirement.

**Reply by ghost** [reply]
![Screenshot from 2022-05-14 15-16-17](https://user-images.githubusercontent.com/91938769/168517968-c2c5741c-7ad7-4bfb-878e-5572e2b5a853.png)
![Screenshot from 2022-05-14 15-15-36](https://user-images.githubusercontent.com/91938769/168517992-c811ef16-da89-43b3-833e-f1ca5e3d7976.png)

Hii @sa-si-dev .
Thank you for replying back to me. Actually I solved the issue.
Here is the code change that I made  in dist/virtual-select.js file (line 2008)
<img width="626" alt="Screenshot 2022-05-16 at 9 36 51 AM" src="https://user-images.githubusercontent.com/91938769/168518144-4cdeb94f-ee96-482f-afdc-ccd5accd0f75.png">

**Reply by ghost** [reply]
Actually this is an autocomplete kind of feature that was a requirement in my organization.
If you allow me can I please contribute to your project.
@sa-si-dev

**Reply by sa-si-dev** [reply]
> Actually this is an autocomplete kind of feature that was a requirement in my organization.
> If you allow me can I please contribute to your project.
> @sa-si-dev 

You are welcome for the contribution.

**Comment by sa-si-dev** [comment, marked as answer]
@gouravojhaAntino Set the property `searchByStartsWith: true` to enable this feature.

This feature is available in the [release v1.0.30](https://github.com/sa-si-dev/virtual-select/releases/tag/v1.0.30)

**Reply by bunajianke** [reply]
when i use `search: true` `searchByStartsWith: true` feature, i need to search the result by `value-key`, but it only search `label-key`.
@sa-si-dev

**Reply by sa-si-dev** [reply]
> when i use `search: true` `searchByStartsWith: true` feature, i need to search the result by `value-key`, but it only search `label-key`.
> @sa-si-dev 

@bunajianke To include different value for searching, use `aliasKey: PROP_NAME` property.

In your case, it would be `aliasKey: 'value'`.

**Reply by bunajianke** [reply]
Let me try, thanks
